### ISSUE TITLE
It should be possible to use system packages for `test_dist`

### ISSUE THREAD

**Issue by hakuch** [scylladb/seastar#592]
Observed with: 5fc32fe61636821f8efa73d601c4587fe8759c56

The `test_dist` target verifies that consumers can use Seastar correctly.

We don't want to require a particular platform for running the test, so we currently use cmake-cooking to fetch and prepare all of Seastar's dependencies in a platform-agnostic way.

This works well, but the test takes a long time to execute the first time since all the dependencies have to be built (in subsequent executions, everything is cached).

As a result, we don't currently run `test_dist` automatically with `test.py` and by extension in CI.

Some build issues could have been avoided if we *did* automatically run `test_dist`.

If we know ahead of time that we're on a platform where system packages are available, then it should be possible to optionally run `test_dist` without cmake-cooking in order to save time and disk space.

**Comment by espindola** [comment]
Using available system packages and cooking only for what is missing would be the perfect behaviour IMHO.

Other than the convenience, that would make the test test what should be configure's default behaviour.

**Comment by hakuch** [comment]
I agree. However, that's a change to `configure.py` and not to `test_dist` itself.
